velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -65,6 +65,51 @@ std::shared_ptr<HiveBucketProperty> buildHiveBucketProperty(
       bucketTypes,
       sortBy);
 }
+
+core::IndexJoinConditionPtr parseJoinCondition(
+    const std::string& joinCondition,
+    const RowTypePtr& rowType,
+    const parse::ParseOptions& options,
+    memory::MemoryPool* pool) {
+  const auto unTypedJoinCondition = parse::parseExpr(joinCondition, options);
+  const auto unTypedCallExpr =
+      std::dynamic_pointer_cast<const core::CallExpr>(unTypedJoinCondition);
+  VELOX_CHECK_NOT_NULL(unTypedCallExpr);
+  if (unTypedCallExpr->getFunctionName() == "contains") {
+    VELOX_CHECK_EQ(unTypedCallExpr->getInputs().size(), 2);
+    auto keyColumnExpr =
+        std::dynamic_pointer_cast<const core::FieldAccessTypedExpr>(
+            core::Expressions::inferTypes(
+                unTypedCallExpr->getInputs()[0], rowType, pool));
+    VELOX_CHECK_NOT_NULL(keyColumnExpr);
+    auto conditionColumnExpr =
+        std::dynamic_pointer_cast<const core::FieldAccessTypedExpr>(
+            core::Expressions::inferTypes(
+                unTypedCallExpr->getInputs()[1], rowType, pool));
+    VELOX_CHECK_NOT_NULL(conditionColumnExpr);
+    return std::make_shared<core::InIndexJoinCondition>(
+        std::move(keyColumnExpr), std::move(conditionColumnExpr));
+  }
+
+  if (unTypedCallExpr->getFunctionName() == "between") {
+    VELOX_CHECK_EQ(unTypedCallExpr->getInputs().size(), 3);
+    auto keyColumnExpr =
+        std::dynamic_pointer_cast<const core::FieldAccessTypedExpr>(
+            core::Expressions::inferTypes(
+                unTypedCallExpr->getInputs()[0], rowType, pool));
+    VELOX_CHECK_NOT_NULL(keyColumnExpr);
+    auto lowerExpr = core::Expressions::inferTypes(
+        unTypedCallExpr->getInputs()[1], rowType, pool);
+    VELOX_CHECK_NOT_NULL(lowerExpr);
+    auto upperExpr = core::Expressions::inferTypes(
+        unTypedCallExpr->getInputs()[1], rowType, pool);
+    VELOX_CHECK_NOT_NULL(upperExpr);
+    return std::make_shared<core::BetweenIndexJoinCondition>(
+        std::move(keyColumnExpr), std::move(lowerExpr), std::move(upperExpr));
+  }
+
+  VELOX_USER_FAIL("Invalid index join condition: {}", joinCondition);
```

Comment:
Would you update the API comments to clarify what are supported conditions and enhance this error message to include that as well?